android/player/src/main/java/com/intuit/playerui/android/asset/SuspendableAsset.kt

```diff
@@ -46,17 +57,71 @@ public abstract class SuspendableAsset<Data>(assetContext: AssetContext, seriali
     final override fun View.hydrate() {
         if (this is AsyncViewStub) return
 
+        player.asyncHydrationTrackerPlugin?.trackHydration(this@SuspendableAsset)
         setTag(R.bool.view_hydrated, false)
         hydrationScope.launch(Dispatchers.Main) { doHydrate() }
     }
 
     private suspend fun View.doHydrate() = withContext(Dispatchers.Main) {
+        player.asyncHydrationTrackerPlugin?.trackHydration(this@SuspendableAsset)
```

Comment:
what is the reason for calling this everywhere

Reply:
There are several paths towards hydration:
- `initView` -> `doHydrate`
- `rehydrate` -> `hydrate`
- `initView` -> `hydrate`

We call this at each entry point to ensure that tracking is started as soon as possible, in a synchronous manner. I think the `doHydrate` on might be redundant. It was the original location, but we introduced to `initView` to ensure we started tracking hydration _before_ we launch the coroutine to create anything.